### ISSUE TITLE
Latest Betaflight firmware : Baro and Mag not found for GY-91 and mixer mode issue in "motors"

### ISSUE THREAD

**Issue by jct1411** [rizacelik/STM32F405RGT6_INAV_Firmware#16]
Dear Sir,

I have setup some config with WeAct Studio STM32F405RGT6 Board on a breadboard with GY-91 only.  I flashed the board with the latest Betaflight firmware of 01 dec 2024.  Gyro and Accel are well recognized but not Baro nor Mag ?  Baro and Mag of the same physical GY-91 are well recognized in a similar configuration with Betaflight firmware 4.5.0 on STM32F411

Also going into "motors" there is popup with "Mixer mode problem detected".  Looking at resources with CLI, I only see Motor 1 and Motor 2.  I do not see anything for Motor 3 and Motor 4 (B06 and B07 are also marked as free)

![STM32F405RGT6](https://github.com/user-attachments/assets/a2642587-fcda-409d-b298-4e763ef1b2d8)
![Betaflight_MixerMode](https://github.com/user-attachments/assets/a1e96d2a-594a-4f9e-ac73-cd12999f5cd6)
![Resources](https://github.com/user-attachments/assets/de94b0a6-de5e-416e-ad37-0223c2dae788)

Thanks for your help

**Comment by jct1411** [comment]
Strange, same physical GY-91 module with STM32F411 shows Baro (not Mag).  I only have one GY-87 module where the compass is working ...   I will check again the connections for GY-91

Strange that I do not find the resources for Motor 3 and Motor 4.  Can I add these manually ?  Why don't they appear automatically after loading the firmware ?

**Comment by rizacelik** [comment]
You are right, there is a conflict between MOTOR 1 TIMER and DMA and it has been resolved. The connection diagram has been recreated. You can test it.

**Comment by jct1411** [comment]
Thank you for the modification : popup does no more appear when selecting "motors".  However I don't measure any signal on any of the motors' pins.

**Comment by rizacelik** [comment]
You can test it by giving output on this screen. You should get a signal.

![image](https://github.com/user-attachments/assets/d924c355-3aea-443a-869d-47e810d21742)

**Comment by jct1411** [comment]
This is the method I have used but with PMW.  I have again tested this method on another configuration with STM32F411 and I immediately see the signals but on the breadboard configuration with STM32F405RGT6 no signal is seen (I use an oscilloscope).  There must be an issue somewhere ...

**Comment by rizacelik** [comment]
Problem fixed. All motors are working. Download and testing.

**Comment by jct1411** [comment]
Motor 1, Motor 2 Motor 3 are working .  **However Motor 4 does not**.  I see Motor 4 is the only one using TIM3 CH2.  May be there is the problem.  I rememeber a few months ago having looked at such an issue but I can no more remember what exactly the issue was.  I found this by lookink at the technical documentation of the processor but I do no more find it ...

I also tried UART 1 with GPS.  It doesn't seem to work.    Betaflight does not detect the GPS when it is connected (icon GPS does not appear, even red when just connected).  Il will check the GPS on another configuration with STM32F411

**Comment by jct1411** [comment]
I confirm indeed there seems to be an issue with GPS UART 1 since ithe same physical GPS works on ST32M32F411 config but not on STM32F405RGT6 config ...

**Comment by rizacelik** [comment]
I disabled the servos, 4 motors work but the GPS did not work. I tried very hard but it did not work. I am removing the firmware. I will recompile it at a later time.

**Comment by jct1411** [comment]
Thanks a lot for all the work already done.  I know how this can be time consuming.
Waiting for a new firmware version that I will test as soon as available.

**Comment by Flighthobbystm** [comment]
> Problem fixed. All motors are working. Download and testing.

Where can I download Betaflight ?? fixed??
